src/grid.jl

```diff
@@ -0,0 +1,90 @@
+# Grid types
+export Node, Cell, Grid
+
+# Cell typealias
+export Line, QuadraticLine,
+       Triangle, QuadraticTriangle, Quadrilateral, QuadraticQuadrilateral,
+       Tetrahedron, QuadraticTetrahedron, Hexahedron, QuadraticHexahedron
+
+# Grid utilities
+export cells, nodes, cell_coordinates, n_cells, n_nodes
+
+#########################
+# Main types for meshes #
+#########################
+"""
+A `Node` is a point in space.
+"""
+immutable Node{dim, T}
+    x::Vec{dim, T}
+end
+
+"""
+A `Cell` is a sub-domain defined by a collection of `Node`s as vertices.
+"""
+immutable Cell{dim, N}
+    id::Int
```

Comment:
Yea thats nice, although, the way `FEBoundaryValues` got implemented we need all coordinates of the cell for the boundary too (for `reinit!`), so for the case of `coordinates`, the implementation would be the same